### PR TITLE
Fix example in gems/contract-programming

### PR DESCRIPTION
**Pull request by wilzbach**
The invariant will be run directly after the constructor. I took the opportunity and added an example for `out` (that combines `toString`) and stressed the importance of using `enforce` for user input

### CODE REVIEW

File: public/content/en/gems/contract-programming.md [hunk 1]
```diff
@@ -96,17 +109,34 @@ struct Date {
...
     }
+
+    // Serializes Date object to YYYY-MM-DD
+	string toString() const
+    out (result) {
+        import std.algorithm: all, count, equal, map;
+        import std.string: isNumeric;
+        import std.array: split;
+
+		// verify we return YYYY-MM-DD
```
Comment (stonemaster) [inline public/content/en/gems/contract-programming.md]: one space more indent

File: public/content/en/gems/contract-programming.md [hunk 2]
```diff
@@ -96,17 +109,34 @@ struct Date {
...
+    auto date = Date(2016, 2, 7);
+
     // This will make invariant fail.
-    // Of course this should not be validated
-    // inside a invariant! External data needs
-    // to be checked in release mode too, and
-    // be propgated through e.g. exceptions.
-    date.fromString("2016-13-7");
+    // Don't validate user input with contracts,
+    // throw exceptions instead.
```
Comment (stonemaster) [inline public/content/en/gems/contract-programming.md]: indent not aligned

File: public/content/en/gems/contract-programming.md [hunk 3]
```diff
@@ -96,17 +109,34 @@ struct Date {
...
+        import std.algorithm: all, count, equal, map;
+        import std.string: isNumeric;
+        import std.array: split;
+
+		// verify we return YYYY-MM-DD
+        assert(result.count("-") == 2);
+        auto parts = result.split("-");
+        assert(parts.map!`a.length`.equal([4, 2, 2]));
+        assert(parts.all!isNumeric);
+    }
```
Comment (wilzbach) [inline public/content/en/gems/contract-programming.md]: Was an issue with tabs in the markdown file - maybe we should forbid them completely there?

**Comment by stonemaster** [comment]
Line length error

**Comment by wilzbach** [comment]
> Line length error

was fixed, forgot to comment

File: public/content/en/gems/contract-programming.md [hunk 4]
```diff
@@ -50,43 +51,55 @@ state during its whole lifetime:
 * `invariant()` is called after exiting a member
   function.
 
+### Validating user input
+
+As all contracts will be removed in the release build, user input should not
+be checked using contracts. Moreover `assert` can be in `nothrow` functions,
```
Comment (stonemaster) [inline public/content/en/gems/contract-programming.md]: "can still be used in"..

**Comment by stonemaster** [comment]
Reopening to let travis run

**Comment by stonemaster** [comment]
Libevent error which has been fixed in master. mErging as it passed before.

**Comment by wilzbach** [comment]
> Reopening to let travis run

I rebased, maybe that helps ;-)

**Pull request merged**